### PR TITLE
Add interface support for qualified_next_hop

### LINKED ISSUE #71: Add interface option to qualified next hop on static route
Missing the `interface` option on the `qualified_next_hop` portion of the static route resource.

Would allow a command like this to be done with resource.

```
set routing-instances test routing-options rib DATA.inet6.0 static route 1234:1234::/32 qualified-next-hop 1234:1234:11:1::1 interface ge-0/0/0.123
```

### PR DESCRIPTION
**Pull request by tagur87**
This allows specifying the interface for a qualified next hop in addition to metric, preference, and next_hop options on the static route resource

ENHANCEMENTS:

* add `interface` option to `qualified_next_hop` on `static_route` resource (Fixes #71)

### CODE REVIEW

**Comment by tagur87** [comment]
@jeremmfr - Ready for your review

File: junos/resource_static_route_test.go [hunk 1]
```diff
@@ -147,10 +187,23 @@ resource junos_static_route testacc_staticRoute_instance {
   qualified_next_hop {
     next_hop   = "st0.0"
     preference = 101
-    metric     = 101
+	metric     = 101
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:190]: Unnecessary change

File: junos/resource_static_route_test.go [hunk 2]
```diff
@@ -147,10 +187,23 @@ resource junos_static_route testacc_staticRoute_instance {
...
+  preference       = 102
+  metric           = 102
+  qualified_next_hop {
+	next_hop   = "192.0.1.1"
+	preference = 102
+	metric     = 102
+	interface  = "st0.0"
+}
+community = ["no-advertise"]
+}
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:206]: Unnecessary new resource. You can just add a second `qualified_next_hop` to the existing resource with only arguments to test : 
```
 qualified_next_hop {
  next_hop   = "192.0.2.250"
  interface  = "st0.0"
}
```
A second resource will extend the test time.

File: junos/resource_static_route_test.go [hunk 3]
```diff
@@ -188,6 +241,19 @@ resource junos_static_route testacc_staticRoute_ipv6_instance {
...
+  preference       = 102
+  metric           = 102
+  qualified_next_hop {
+	next_hop = "2001:db8:85a4::1"
+	preference = 102
+	metric = 102
+	interface = "st0.0"
+  }
+  community = ["no-advertise"]
+}
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:256]: Same comment as for ipv4. Just add second `qualified_next_hop` to the existing resource

File: junos/resource_static_route_test.go [hunk 4]
```diff
@@ -38,6 +38,26 @@ func TestAccJunosStaticRoute_basic(t *testing.T) {
...
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_interface",
+							"qualified_next_hop.0.preference", "102"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_interface",
+							"qualified_next_hop.0.metric", "102"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_interface",
+							"qualified_next_hop.0.interface", "st0.0"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_interface",
+							"community.#", "1"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_interface",
+							"community.0", "no-advertise"),
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:60]: Unnecessary tests, several arguments already tested. Just add the test of the `interface` argument.

File: junos/resource_static_route_test.go [hunk 5]
```diff
@@ -80,6 +100,26 @@ func TestAccJunosStaticRoute_basic(t *testing.T) {
...
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_interface",
+							"qualified_next_hop.0.preference", "102"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_interface",
+							"qualified_next_hop.0.metric", "102"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_interface",
+							"qualified_next_hop.0.interface", "st0.0"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_interface",
+							"community.#", "1"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_interface",
+							"community.0", "no-advertise"),
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:122]: Same comment as for ipv4. Just test `interface` argument

**Review by jeremmfr — CHANGES_REQUESTED** [review]
Can you fix the changes in the go test ?

File: junos/resource_static_route_test.go (hunk 1 above)
Comment (tagur87) [reply junos/resource_static_route_test.go]: Removed this erroneous change

File: junos/resource_static_route_test.go (hunk 2 above)
Comment (tagur87) [reply junos/resource_static_route_test.go]: Fixed

File: junos/resource_static_route_test.go (hunk 4 above)
Comment (tagur87) [reply junos/resource_static_route_test.go]: Fixed

File: junos/resource_static_route_test.go (hunk 5 above)
Comment (tagur87) [reply junos/resource_static_route_test.go]: Fixed

File: junos/resource_static_route_test.go (hunk 3 above)
Comment (tagur87) [reply junos/resource_static_route_test.go]: Fixed

File: junos/resource_static_route_test.go [hunk 6]
```diff
@@ -69,13 +73,17 @@ func TestAccJunosStaticRoute_basic(t *testing.T) {
...
 						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_instance",
 							"qualified_next_hop.0.next_hop", "st0.0"),
 						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_instance",
 							"qualified_next_hop.0.preference", "101"),
 						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_instance",
 							"qualified_next_hop.0.metric", "101"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_instance",
+							"qualified_next_hop.1.next_hop", "2001:db8:85a4::1"),
+						resource.TestCheckResourceAttr("junos_static_route.testacc_staticRoute_ipv6_instance",
+							"qualified_next_hop.1.preference", "st0.0"),
```
Comment (jeremmfr) [inline junos/resource_static_route_test.go:86]: it should be `qualified_next_hop.1.interface`

**Review by jeremmfr — APPROVED** [review]
There is a minor error but I will correct it myself

File: junos/resource_static_route_test.go (hunk 6 above)
Comment (tagur87) [reply junos/resource_static_route_test.go:86]: Was trying to fix too late last night! :wink:

Thanks for the help!
